src/setuptools_scm/_entrypoints.py

```diff
@@ -17,15 +16,32 @@
     from ._config import Configuration, ParseFunction
 
 
-log = _log.log.getChild("entrypoints")
+from importlib.metadata import EntryPoint as EntryPoint
+
+
+if sys.version_info[:2] < (3, 10):
+    from importlib.metadata import entry_points as legacy_entry_points
+
+    class EntryPoints:
+        _groupdata: list[EntryPoint]
+
+        def __init__(self, groupdata: list[EntryPoint]) -> None:
+            self._groupdata = groupdata
+
+        def select(self, name: str) -> EntryPoints:
+            return EntryPoints([x for x in self._groupdata if x.name == name])
```

Comment:
nice, matches pretty close to [3.11 impl](https://github.com/python/cpython/blob/3.11/Lib/importlib/metadata/__init__.py#L375C1-L380C70) of this